### PR TITLE
optimize code structure

### PR DESCRIPTION
**Pull request by du2016**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
/kind cleanup
> /kind design
> /kind documentation
> /kind test
> /kind failing-test
> /kind feature

**What this PR does / why we need it**:
optimize code structure
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by fisherxu** [comment]
LGTM, Thanks for the nice optimization! There's some vendor check errs now, could you plz run `hack/update-vendor.sh` to update the vendor files? @du2016

File: cloud/pkg/cloudhub/servers/server.go [hunk 1]
```diff
@@ -37,13 +37,13 @@ func StartCloudHub(config *util.Config, messageq *channelq.ChannelMessageQueue)
 	handler.InitHandler(config, messageq)
 
 	svc := server.Server{
-		Type:       api.ProtocolTypeQuic,
+		Type:       protocolType,
 		Addr:       fmt.Sprintf("%s:%d", config.Address, config.QuicPort),
 		TLSConfig:  &tlsConfig,
 		AutoRoute:  true,
 		ConnNotify: handler.CloudhubHandler.OnRegister,
 		ExOpts:     api.QuicServerOption{MaxIncomingStreams: config.MaxIncomingStreams},
```
Comment (kadisi) [inline cloud/pkg/cloudhub/servers/server.go]: for websocket, it is `api.WSServerOption{Path: "/"},`

File: cloud/pkg/cloudhub/servers/server.go [hunk 2]
```diff
@@ -37,13 +35,13 @@ func StartCloudHub(config *util.Config, messageq *channelq.ChannelMessageQueue)
...
 	svc := server.Server{
-		Type:       api.ProtocolTypeQuic,
+		Type:       protocolType,
 		Addr:       fmt.Sprintf("%s:%d", config.Address, config.QuicPort),
 		TLSConfig:  &tlsConfig,
 		AutoRoute:  true,
 		ConnNotify: handler.CloudhubHandler.OnRegister,
-		ExOpts:     api.QuicServerOption{MaxIncomingStreams: config.MaxIncomingStreams},
+		ExOpts:     exOpts,
 	}
```
Comment (fisherxu) [inline cloud/pkg/cloudhub/servers/server.go]: We can add a switch logic here instead of pass as parameter. @du2016 @kadisi  Thought?

Comment (du2016) [reply cloud/pkg/cloudhub/servers/server.go]: if add a switch logic,whether should add config parameter to hubconfig?

File: cloud/pkg/cloudhub/servers/server.go (hunk 1 above)
Comment (du2016) [reply cloud/pkg/cloudhub/servers/server.go]: done

File: cloud/pkg/cloudhub/servers/server.go (hunk 2 above)
Comment (kadisi) [reply cloud/pkg/cloudhub/servers/server.go]: @fisherxu  yes , adding a switch logic in `StartCloudHub()` function looks good.

**Comment by kadisi** [comment]
@du2016 please exec `make verify` ,There's some vendor check errs now.

File: cloud/pkg/cloudhub/servers/server.go (hunk 2 above)
Comment (fisherxu) [reply cloud/pkg/cloudhub/servers/server.go]: > whether should add config parameter to hubconfig 

No need add to hubconfig, if add to hubconfig, we still need to judge which protocol used.

Comment (du2016) [reply cloud/pkg/cloudhub/servers/server.go]: OK ,i'll add a switch logic

Comment (fisherxu) [reply cloud/pkg/cloudhub/servers/server.go]: ok, thanks :) then let's got merged.

File: cloud/pkg/cloudhub/servers/server.go [hunk 3]
```diff
@@ -37,13 +37,22 @@ func StartCloudHub(config *util.Config, messageq *channelq.ChannelMessageQueue)
 	handler.InitHandler(config, messageq)
 
 	svc := server.Server{
-		Type:       api.ProtocolTypeQuic,
+		Type:       protocolType,
 		Addr:       fmt.Sprintf("%s:%d", config.Address, config.QuicPort),
```
Comment (fisherxu) [inline cloud/pkg/cloudhub/servers/server.go]: The `QuicPort` also should move to switch :) @du2016

Comment (du2016) [reply cloud/pkg/cloudhub/servers/server.go]: sorry, i'll change it

**Comment by fisherxu** [comment]
/lgtm
Thanks!

**Pull request merged**